src/pykeen/trackers/tensorboard.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+"""An adapter for TensorBoard."""
+
+import datetime
+import pathlib
+from typing import Any, Dict, Mapping, Optional, TYPE_CHECKING, Type
+
+from .base import ResultTracker
+from ..constants import PYKEEN_LOGS
+from ..utils import flatten_dictionary
+
+if TYPE_CHECKING:
+    import torch.utils.tensorboard
+
+__all__ = [
+    'TensorBoardResultTracker',
+]
+
+
+class TensorBoardResultTracker(ResultTracker):
+    """A tracker for TensorBoard."""
+
+    #: The class that's used to instantiate a summarywriter
+    summary_writer_cls: Type['torch.utils.tensorboard.SummaryWriter']
+
+    def __init__(
+        self,
+        experiment_path: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        tags: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initialize result tracking via Tensorboard.
+
+        :param experiment_path:
+            The experiment path. A custom path at which the tensorboard logs will be saved.
+        :param experiment_name:
+            The name of the experiment, will be used as a sub directory name for the logging. If no default is given,
+            the current time is used. If set, experiment_path is set, this argument has no effect.
+        :param tags:
+            The additional run details which are presented as tags to be logged
+        """
+        import torch.utils.tensorboard
+        self.summary_writer_cls = torch.utils.tensorboard.SummaryWriter
+        self.tags = tags
+
+        if experiment_path is None:
+            if experiment_name is None:
+                experiment_name = datetime.datetime.now().isoformat()
```

Comment:
```suggestion
                experiment_name = datetime.datetime.now().isoformat().replace(":", "_")
```

Directly using [datetime.datetime.isoformat](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat) as filename will result in a name such as `2019-05-18T15:17:08.132263` including `:`, which is a reserved character under Windows (cf. [here](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN#file_and_directory_names)).

Reply:
Should I hold off on committing this then?